frontend/app/login.tsx

```diff
@@ -1,12 +1,233 @@
-import { Text, View } from "react-native";
-import React from "react";
+import React, { useState } from "react";
+import { useRouter } from "expo-router";
+import {
+  View,
+  Text,
+  TextInput,
+  TouchableOpacity,
+  StyleSheet,
+} from "react-native";
+import axios from "axios";
 
-const Register = () => {
-  return (
-    <View>
-      <Text>Register</Text>
+import * as SecureStore from "expo-secure-store";
+
+async function save(key: string, value: string) {
+  await SecureStore.setItemAsync(key, value);
+}
+
+function LoginPage() {
+  const router = useRouter();
+  const [isSignUp, setSignUp] = useState(false);
+  const [name, setName] = useState("");
+  const [password, setPassword] = useState("");
+  const [email, setEmail] = useState("");
+  const [phone, setPhone] = useState("");
+  const [toggle, setToggle] = useState(false);
+  const [key, onChangeKey] = React.useState("authtoken");
+  const [value, onChangeValue] = React.useState("Your value here");
+
+  const handleSignUp = async () => {
+    //alert("name: " + username + " password: " + password + " email: " + email + " phone: " + phone);
+
+    try {
+      if (
+        name.length > 0 &&
+        password.length > 0 &&
+        email.length > 0 &&
+        phone.length > 0
+      ) {
+        const response = await axios.post(
+          "https://open-recipes.onrender.com/auth/sign-up",
+          {
+            name: name,
+            email: email,
+            phone: phone,
+            password: password,
+          },
+          {
+            headers: {
+              "Content-Type": "application/json",
+            },
+          },
+        );
+        //alert("response: " + response.status);
+
+        if (response.status === 200 || response.status === 201) {
+          // Registration successful, redirect to login
```

Comment:
I like comments describing how the code will respond to the post requests, it really helped me understand what is going on!